app/services/idv/steps/ipp/ssn_step.rb

```diff
@@ -18,19 +18,26 @@ def call
         def extra_view_variables
           {
             updating_ssn: updating_ssn,
+            threatmetrix_session_id: generate_threatmetrix_session_id,            
           }
         end
 
         private
 
-        def ssn
-          flow_params[:ssn]
-        end
-
         def form_submit
           Idv::SsnFormatForm.new(current_user).submit(permit(:ssn))
         end
 
+        def generate_threatmetrix_session_id
+          return unless IdentityConfig.store.proofing_device_profiling_collecting_enabled
+          flow_session[:threatmetrix_session_id] = SecureRandom.uuid if !updating_ssn
+          flow_session[:threatmetrix_session_id]
+        end
```

Comment:
(not a blocker)

do you think that a `ThreatMetrixHelper` or something mixin module would be a good way to share this implementation across the two steps? then we could write these once

Reply:
Yes, I think that's a good idea

Reply:
one problem with mixins I've seen is it's hard to tell when a method comes from a mixin, so for like this, if we called it `ThreatMetrixStepHelper` and luckily this method name has `_threatmetrix_` in it, so hopefully that is a hint to future code spelunkers